src/js/calculator.js

```diff
@@ -0,0 +1,101 @@
+import { Operators, OperatorSet, NumberSet, ErrorMessages } from './constants.js'
+
+const calculateFunctions = {
+  [Operators.Add]: (operand1, operand2) => operand1 + operand2,
+  [Operators.Subtract]: (operand1, operand2) => operand1 - operand2,
+  [Operators.Multiply]: (operand1, operand2) => operand1 * operand2,
+  [Operators.Divide]: (operand1, operand2) => {
+    if (operand2 === 0) throw new Error(ErrorMessages.DivideZero)
+    return Math.trunc(operand1 / operand2)
+  },
+}
+
+const isNumber = val => NumberSet.has(val)
+const isOperator = val => OperatorSet.has(val)
+
+export default class Calculator {
+  #values = ['0']
+
+  get #lastOperatorIndex() {
+    for (let i = this.#values.length - 1; i >= 0; i--) {
+      const val = this.#values[i]
+      if (OperatorSet.has(val)) return i
+    }
+    return -1
+  }
+  get #isNegative() {
+    return this.#values[this.#lastOperatorIndex] === '-'
+  }
+  get #prevValueSize() {
+    return this.#values.length - (this.#lastOperatorIndex + 1)
+  }
+  #removeFirstZero() {
+    if (
+      (!this.#isNegative && this.#values.length === 1 && this.#values[0] === '0') ||
+      (this.#isNegative && this.#values.length === 2 && this.#values[1] === '0') ||
+      (this.#values.length === this.#lastOperatorIndex + 2 && this.#values[this.#lastOperatorIndex + 1] === '0')
+    ) {
+      this.#values.pop()
+    }
+  }
```

Comment:
```suggestion
  #removeFirstZero() {
    const  isFirstOperandPositiveZero =  !this.#isNegative && this.#values.length === 1 && this.#values[0] === '0';
    const isFirstOperandNegativeZero = this.#isNegative && this.#values.length === 2 && this.#values[1] === '0';
    const isSecondOperandZero=  this.#values.length === this.#lastOperatorIndex + 2 && this.#values[this.#lastOperatorIndex + 1] === '0'; 
    
    if (isFirstOperandPositiveZero || isFirstOperandNegativeZero || isSecondOperandZero) {
      this.#values.pop()
    }
  }
```

이런 식으로 조건에 네이밍을 해주면 가독성이 더 좋아지지 않을까 고민해봤습니다!

Reply:
👍